### DISCUSSION TITLE
tick data to console

### DISCUSSION THREAD

**Discussion by foxaal** [stoqey/ib#82] [Q&A]
Is it possible to print tick data to the console with TWS API?  We are trying to do that with the java version of the TWS API, but no luck yet.

**Comment by mfrener** [comment]
You mean something like `/dist/tools/market-data.js -symbol=AMZN -sectype=STK -exchange=SMART -conid=369193` ?
(run as `node ./node_modules/@stoqey/ib/dist/tools/market-data.js -symbol=AMZN -sectype=STK -exchange=SMART -conid=369193` from your app root).
It will print a JSON with AMZN market data to console.
That's the code https://github.com/stoqey/ib/blob/master/src/tools/market-data.ts

**Reply by foxaal** [reply]
Well, we want to see a stream of live level one ticks coming over the console.  As in data spewing out onto the console for a given ticker symbol.

**Reply by foxaal** [reply]
You can see those ticks on TWS, and it seems we should be able to access those from the API, but no luck so far in Java.  Will switch to this wrapper if this code can do it.

**Reply by foxaal** [reply]
YES!  looking at your code, it looks that will spray data to the console -- yes?

**Reply by mfrener** [reply]
1) clone this repo
2) > yarn install
3) > yarn build
4) (adapt -port of your use IB Gateway or paper account)
 > node .\dist\tools\market-data.js -symbol=AMZN -sectype=STK -exchange=SMART -conid=3691937 -port=7496
[
  [
    "BID",
    3249.56
  ]
]
[
  [
    "BID_SIZE",
    4
  ]
]
[
  [
    "ASK",
    3250.74
  ]
]
[
  [
    "ASK_SIZE",
    1
  ]
]
[
  [
    "LAST",
    3250.23
  ]
]
[
  [
    "LAST_SIZE",
    2
  ]
]
[
  [
    "BID_SIZE",
    4
  ]
]
[
  [
    "ASK_SIZE",
    1
  ]
]
[
  [
    "LAST_SIZE",
    2
  ]
]
[
  [
    "VOLUME",
    11891
  ]
]
[
  [
    "HIGH",
    3259.68
  ]
]
[
  [
    "LOW",
    3236.43
  ]
]
[
  [
    "CLOSE",
    3231.8
  ]
]
[
  [
    "OPEN",
    3245.29
  ]
]
[
  [
    "HALTED",
    0
  ]
]
[
  [
    "BID",
    3249.57
  ]
]
[
  [
    "BID_SIZE",
    1
  ]
]
[
  [
    "ASK",
    3250.75
  ]
]
[
  [
    "ASK_SIZE",
    3
  ]
]
[
  [
    "BID_SIZE",
    1
  ]
]
[
  [
    "ASK_SIZE",
    3
  ]
]
[
  [
    "HALTED",
    0
  ]
]
[
  [
    "BID",
    3249.56
  ]
]
[
  [
    "BID_SIZE",
    2
  ]
]
[
  [
    "BID_SIZE",
    2
  ]
]
[
  [
    "ASK",
    3250.74
  ]
]
[
  [
    "ASK_SIZE",
    1
  ]
]
[
  [
    "ASK_SIZE",
    1
  ]
]
[
  [
    "ASK",
    3250.75
  ]
]
[
  [
    "ASK_SIZE",
    3
  ]
]
[
  [
    "ASK_SIZE",
    3
  ]
]
...

**Reply by foxaal** [reply]
Muchas gracias!
